### PR TITLE
Check Backend service by name instead of path in GCE LB code.

### LINKED ISSUE #96513: GCE L4 ILB forwarding rules getting recreated upon master restart
**What happened**:
I created a GCE k8s cluster from HEAD and a LoadBalancer service with "Internal" annotation. Restarting the master VM deletes and recreates the forwarding rule after startup, even if nothing changed in the service.

**What you expected to happen**:
The forwarding rule should not be deleted.

**How to reproduce it (as minimally and precisely as possible)**:
Create a k8s cluster via kubetest and create an ILB service. Restart the master VM after the service has gotten IP assignment.

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`): 1.20
- Cloud provider or hardware configuration: 
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/sig network
/kind bug

### PR DESCRIPTION
**Pull request by prameshj**
**What type of PR is this?**

**What this PR does / why we need it**:
This adds logging to the ILB ensure code in GCE.
**Which issue(s) this PR fixes**:

Fixes #96513

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by MrHohn — APPROVED** [review]
/lgtm
/approve

**Comment by prameshj** [comment]
Failures might be related to https://github.com/kubernetes/test-infra/issues/19910

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go [hunk 1]
```diff
@@ -979,11 +982,16 @@ func (g *Cloud) ensureInternalForwardingRule(existingFwdRule, newFwdRule *comput
 }
 
 func forwardingRulesEqual(old, new *compute.ForwardingRule) bool {
+	// basepath could have differences like compute.googleapis.com vs www.googleapis.com, compare resourceIDs
+	oldResourceID, err := cloud.ParseResourceURL(old.BackendService)
+	klog.Errorf("forwardingRulesEqual(): failed to parse backend resource URL from existing FR, err - %v", err)
```
Comment (bowei) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:987]: * Don't log in predicates like this.
* Don't eat errors silently.

We should either propagate the error up or state that malformed URLs are not equal to anything.
My take would be to propagate back up and have the caller log the error.

Comment (prameshj) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:987]: As discussed offline, I will be making this change in a follow-up.

**Pull request merged**